Stack/Opc.Ua.Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -1414,7 +1414,23 @@ protected virtual async Task InternalValidateAsync(X509Certificate2Collection ce
                     null, null, "SHA1 signed certificates are not trusted.", null, sresult);
             }
 
-            if (!isECDsaSignature)
+            // check if certificate signature algorithm length is sufficient
+            if (isECDsaSignature)
```

Comment:
shouldnt we do a strict enforce of the OPC UA Specified signature alogrithms / hash algorithms?

like:

if ECC-nistP384 then Oids.ECDsaWithSha384
else invalid

if ECC-brainpoolP384r1 then Oids.ECDsaWithSha384
else invalid

Reply:
I think only the lower bounds need to be checked, if somebody chose Sha512 for a smaller key it is not a security risk.

Reply:
@romanett @mregen 
I agree from the security point of view, but checking lower bounds only might not be fully spec compliant.
I've chosen the checks since the spec currently states:

**NistP256/BrainpoolP256r1:**
```
The length shall be 256 or 384 bits.
The required Hash Algorithm is based on the Curve Size in bits as follows:
SHA-256 if bits <= 256
SHA-384 if bits > 256 and <= 384
```
**NistP384/BrainpoolP384r1:**
```
The length shall be at 384 or 521 bits.
The required Hash Algorithm is based on the Curve Size in bits as follows:
SHA-384 if bits > 256 and <= 384
SHA-512 if bits > 384
```